Add FreeBSD/arm{,64} support

danfe opened the pull request:
- Replace `getauxval(3)`, a non-standard glibc extension, with FreeBSD's native `elf_aux_info(3)`
- Return features supported by the specific machine processor architecture rather than the hardware platform, because the latter is too vague on FreeBSD and does not include specific version (generation)

danfe committed:
Add FreeBSD/arm{,64} support:

- Replace getauxval(3), a non-standard glibc extension, with
  FreeBSD's native elf_aux_info(3)
- Return features supported by the specific machine processor
  architecture (uname -p) rather than the hardware platform
  (uname -m), because the latter is too vague on FreeBSD and
  does not include specific version (generation)

mgorny commented on src/hwcap.c:
```diff
@@ -14,13 +14,19 @@
-#	error "Platform not supported (only Linux supported at the moment)"
 #endif
+
 #ifndef HAVE_GETAUXVAL
```
```suggestion
#ifdef __FreeBSD__
```

FWICS `elf_aux_info` is specific to FreeBSD.

danfe replied:
I typically think that it's better to check for features rather than particular operating system or its version.  In this case, I found that `HAVE_GETAUXVAL` is exactly what I need: if no `getauxval()` provided by the system, fallback to `elf_aux_info()` as the only known (to me at this point) alternative.  But I won't insist, if you prefer `#ifdef __FreeBSD__` guard, it's okay, but then you should probably remove the check for `getauxval()` from the configure script.

mgorny replied:
Yes, once this lands I plan to replace the build system with meson and I suppose I can remove the redundant check. I suppose another option would be to check for `getauxval` explicitly but TBH I don't think we can really expect that much portability between different platforms.

danfe committed:
#ifndef HAVE_GETAUXVAL -> #ifdef __FreeBSD__

Address feedback from Michał and replace (to be potentially removed)
check for this particular function, as we cannot really expect that
much portability between different platforms.

mgorny commented:
Thanks!
